diff --git a/menu.c b/menu.c
index c98160e265d178eb1d69c97dc9a630ae77a5c922..a15c3c19f8f63d1ac4f5d9f0f736abf5ae69c2a7 100644
--- a/menu.c
+++ b/menu.c
@@ -1181,10 +1181,10 @@ rescan:
 				break;
 			}
 		}
-		else if (inp & PBTN_MA2) {
-			g_autostateld_opt = !g_autostateld_opt;
-			show_help = 3;
-		}
+		// else if (inp & PBTN_MA2) {
+		// 	g_autostateld_opt = !g_autostateld_opt;
+		// 	show_help = 3;
+		// }
 		else if (inp & PBTN_CHAR) {
 			// must be last
 			// sel = dirent_seek_char(namelist, n, sel, cinp); // crashes with R on trimui smart
@@ -1282,9 +1282,9 @@ static void draw_savestate_menu(int menu_sel, int is_loading)
 	menu_draw_end();
 }
 
-static int menu_loop_savestate(int is_loading)
+static int menu_loop_savestate(int is_loading, int menu_sel)
 {
-	static int menu_sel = STATE_SLOT_COUNT;
+	// static int menu_sel = STATE_SLOT_COUNT;
 	int menu_sel_max = STATE_SLOT_COUNT;
 	unsigned long inp = 0;
 	int ret = 0;
